lib/src/components/NxPolicyViolationIndicator/__tests__/NxPolicyViolationIndicator.test.tsx

```diff
@@ -4,47 +4,89 @@
+
+    rerender(<NxPolicyViolationIndicator threatLevelCategory='moderate'></NxPolicyViolationIndicator>);
+    expect(container).toHaveTextContent('moderate');
   });
 
-  it('sets the modifier class using the threatLevelCategory if both props are provided', function() {
-    const component = getShallowComponent({ policyThreatLevel: 9, threatLevelCategory: 'low' });
+  it('overrides the default threatLevelCategory text if text is provided', function() {
+    const { container } = render(
+      <NxPolicyViolationIndicator threatLevelCategory='low'>Foo</NxPolicyViolationIndicator>
+    );
 
-    expect(component).toMatchSelector('.nx-policy-violation-indicator--low');
-    expect(component).not.toMatchSelector('.nx-policy-violation-indicator--critical');
-    expect(component).not.toMatchSelector('.nx-policy-violation-indicator--unspecified');
+    expect(container).not.toHaveTextContent('low');
+    expect(container).toHaveTextContent('Foo');
   });
 
-  it('adds aria attrs to help the icon show up for screen readers', function() {
-    const component = getShallowComponent({ policyThreatLevel: 9, threatLevelCategory: 'low' });
+  it('sets threatLevelCategory to unspecified if not defined', function() {
+    const unspecifiedEl = renderEl()!;
+    expect(unspecifiedEl).toHaveTextContent('unspecified');
+  });
+
+  it('sets the threatLevelCategory by converting policyThreatLevel if it is provided',
+      function() {
+        expect(renderEl({ policyThreatLevel: 0})!).toHaveTextContent('none');
+        expect(renderEl({ policyThreatLevel: 1})!).toHaveTextContent('low');
+        expect(renderEl({ policyThreatLevel: 3})!).toHaveTextContent('moderate');
+        expect(renderEl({ policyThreatLevel: 5})!).toHaveTextContent('severe');
+        expect(renderEl({ policyThreatLevel: 9})!).toHaveTextContent('critical');
+        expect(renderEl({ policyThreatLevel: 0})!).toHaveTextContent('none');
+        expect(renderEl()!).toHaveTextContent('unspecified');
+      });
+
+  it('renders an indicator according to threatLevelCategory over policyThreatNumber of both are provided', function() {
+    expect(renderEl({ policyThreatLevel: 9, threatLevelCategory: 'low'})).toHaveTextContent('low');
+  });
+
+  it('adds aria-label to icon', function() {
+    expect(quickRender().getByRole('img')).toHaveAttribute('aria-label', 'threat level unspecified');
+    expect(quickRender({ threatLevelCategory: 'low' }).getByRole('img'))
+        .toHaveAttribute('aria-label', 'threat level low');
+    expect(quickRender({ threatLevelCategory: 'severe'}).getByRole('img'))
+        .toHaveAttribute('aria-label', 'threat level severe');
+  });
 
-    expect(component.find(NxFontAwesomeIcon)).toHaveProp('aria-label', 'threat level low');
-    expect(component.find(NxFontAwesomeIcon)).toHaveProp('aria-hidden', false);
+  it('sets aria-hidden to false on icon', function() {
+    expect(quickRender().getByRole('img')).toHaveAttribute('aria-hidden', 'false');
```

Comment:
The `aria-hidden` attribute itself is also an impl detail, we just want to make sure the image isn't hidden.  But there's not reason to think it would be, so this test seems kind of unnecessary to me (as does the presence of the `aria-hidden` attr in the implementation.  At any rate, the `getByRole('img')` calls in the test above ensure that the image is not hidden, as they exclude hidden elements by default.  So with that this test is definitely not necessary.

Reply:
> as does the presence of the `aria-hidden` attr in the implementation

Apparently fontawesome icons have `aria-hidden="true"` by default so that explains that

Reply:
>The aria-hidden attribute itself is also an impl detail, we just want to make sure the image isn't hidden. But there's not reason to think it would be, so this test seems kind of unnecessary to me (as does the presence of the aria-hidden attr in the implementation). At any rate, the getByRole('img') calls in the test above ensure that the image is not hidden, as they exclude hidden elements by default. So with that this test is definitely not necessary.

makes sense, I removed the test here: [c244209](https://github.com/sonatype/sonatype-react-shared-components/pull/911/commits/c24420967b560540051227a848bf623fe0b607ba)